libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -985,12 +985,72 @@ void IRGeneratorForStatements::endVisit(MemberAccess const& _memberAccess)
 	}
 	case Type::Category::TypeType:
 	{
-		TypeType const& type = dynamic_cast<TypeType const&>(*_memberAccess.expression().annotation().type);
-		solUnimplementedAssert(type.actualType()->category() == Type::Category::Contract, "");
-		FunctionType const* funType = dynamic_cast<FunctionType const*>(_memberAccess.annotation().type);
-		solUnimplementedAssert(funType, "");
-		solUnimplementedAssert(funType->kind() == FunctionType::Kind::Declaration, "");
-		// Nothing to do for declaration.
+		Type const& actualType = *dynamic_cast<TypeType const&>(
+			*_memberAccess.expression().annotation().type
+		).actualType();
+
+		if (actualType.category() == Type::Category::Contract)
+		{
+			if (auto variable = dynamic_cast<VariableDeclaration const*>(_memberAccess.annotation().referencedDeclaration))
```

Comment:
I think we should stick to extracting `const`, `&` and `*` from auto, or at least be consistent with it.
Down few lines there is usage like `auto const*`